src/main/java/com/health/controller/HomeController.java

```diff
@@ -4246,6 +4340,173 @@ public String updateSpokenVideoPost(HttpServletRequest req, Model model, Princip
      * Spoken Video End
      *************************************/
 
+    /************************************
+     * Video Resource Start
+     **********************/
+
+    @GetMapping("/addVideoResource")
+    public String addVideoResource(HttpServletRequest req, Principal principal, Model model) {
+        User usr = getUser(principal);
+        logger.info("{} {} {}", usr.getUsername(), req.getMethod(), req.getRequestURI());
+        model.addAttribute("userInfo", usr);
+        Path csvFilePath = Paths.get(env.getProperty("spring.applicationexternalPath.name"),
+                CommonData.uploadDirectorySource, "CSVFile", "Sample_CSV_file_for_Video_Resource" + ".csv");
+        String temp = csvFilePath.toString();
+        int indexToStart = temp.indexOf("Media");
+        String document = temp.substring(indexToStart, temp.length());
+        model.addAttribute("sample_csv_file", document);
+
+        List<VideoResource> videoResourceList = videoResourceService.findAll();
+        Collections.sort(videoResourceList, VideoResource.SortByUploadTime);
+        model.addAttribute("videoResourceList", videoResourceList);
+        return "addVideoResource";
+    }
+
+    @PostMapping("/addVideoResource")
+    public String addVideoResourcePost(@RequestParam(value = "add_csv_file") MultipartFile csv_file,
+            HttpServletRequest req, Principal principal, Model model) {
+
+        User usr = getUser(principal);
+        logger.info("{} {} {}", usr.getUsername(), req.getMethod(), req.getRequestURI());
+        model.addAttribute("userInfo", usr);
+
+        try {
+            videoResourceService.saveVideoResourcesFromCSV(csv_file, model);
+
+        } catch (IOException e) {
+
+            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
+            logger.error("Exception: ", e);
+            return addVideoResource(req, principal, model);
+        } catch (CsvException e) {
+            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
+            logger.error("Exception: ", e);
+            return addVideoResource(req, principal, model);
+        } catch (NoSuchAlgorithmException e) {
+            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
+            logger.error("Exception: ", e);
+            return addVideoResource(req, principal, model);
+        }

```

Comment:
```suggestion
        } catch (Exception e) {
            model.addAttribute("error_msg", "Some Errors Occured Please contact Admin or try again");
            logger.error("Exception: ", e);
            return addVideoResource(req, principal, model);
        }
```

As the exception handling is the same for all exceptions, add a common catch block.